docker/Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM python:3.12
+
+WORKDIR /code
+
+COPY git-manifest.yaml /code/git-manifest.yaml
+COPY docker/logging.yaml /code/logging.yaml
+COPY src/talk2powersystemllm/app/trouble.md /code/trouble.md
+COPY src/talk2powersystemllm/ /code/talk2powersystemllm/
+COPY config/ontology/ /code/config/ontology/
+COPY config/*yaml /code/config/
+
+COPY ./pyproject.toml ./poetry.lock ./README.adoc /code/
+
+ENV POETRY_VIRTUALENVS_CREATE=false \
+    POETRY_NO_INTERACTION=1
+
+RUN pip install poetry==2.1.3
```

Comment:
You could combine all RUNs into a single one like so 

```
RUN command-a && \
    command-b && \
     ....
    command-z
```

This will produce fewer layers in the image.